From 7c34403d02be7ac2557d238e25611ae1a9983c4d Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Sat, 19 Dec 2015 23:53:21 -0800
Subject: [PATCH] QCamera2: Force larger preview size (640x480) for WeChat

WeChat likes to set the preview size to the minimum available (320x240), as
well as the preview frame-rate to 15 FPS. We can identify WeChat by these things
and force it to use a larger preview size for improved video-chat quality and
UX.
---
 QCamera2/HAL/QCameraParameters.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/QCamera2/HAL/QCameraParameters.cpp b/QCamera2/HAL/QCameraParameters.cpp
index 2cfbd9f0..93af4980 100644
--- a/QCamera2/HAL/QCameraParameters.cpp
+++ b/QCamera2/HAL/QCameraParameters.cpp
@@ -1842,6 +1842,14 @@ int32_t QCameraParameters::setPreviewFrameRate(const QCameraParameters& params)
 {
     const char *str = params.get(KEY_PREVIEW_FRAME_RATE);
     const char *prev_str = get(KEY_PREVIEW_FRAME_RATE);
+    int width, height;
+
+    // Force better preview size for WeChat
+    if (!strcmp(str, "15")) {
+        params.getPreviewSize(&width, &height);
+        if (width == 320 && height == 240)
+            CameraParameters::setPreviewSize(640, 480);
+    }
 
     if ( str ) {
         if ( prev_str &&
-- 
2.13.2

